src/build.h

```diff
@@ -92,6 +94,10 @@ struct Plan {
 
   set<Edge*> ready_;
 
+  /// user provided targets in build order, earlier one have higher priority
+  vector<Node*> targets_;
+  list<Edge*> priority_list_;
```

Comment:
Anything stopping you from using a `std::vector` for the `priority_list_`, too?

Reply:
I thought at build.cc line 359

``` C++
  for (list<Edge*>::iterator it = priority_list_.begin(),
                             end = priority_list_.end();
       it != end; ++it) {
    i = ready_.find(*it);
    if (i != ready_.end()) {
      priority_list_.erase(it);
      break;
    }
  }

```
this may remove element from priority_list_ at middle pos , list is more friendly ,and erase don't need a extra resize . (maybe vector is faster for iterate?  I didn't benchmark this. I didn't even notice this changes because I just added minimal logic on original cpsched) , if vector is better choice ,I'll modify this.